### PR TITLE
fix(resolver): consolidate reference resolution 

### PR DESCRIPTION
**Pull request by icecrasher321**
## Summary
Standard system in block resolver class extracted and reused. 

Function / Code resolution bugs fixed. 

Other edge cases like single letter block names, and _ collisions handled. 

## Type of Change
- [x] Bug fix

## Testing
Added tests + tested manually

## Checklist
- [x] Code follows project style guidelines
- [x] Self-reviewed my changes
- [x] Tests added/updated and passing
- [x] No new warnings introduced
- [x] I confirm that I have read and agree to the terms outlined in the [Contributor License Agreement (CLA)](./CONTRIBUTING.md#contributor-license-agreement-cla)

### CODE REVIEW

**Comment by icecrasher321** [comment]
@cursor review

**Comment by icecrasher321** [comment]
@cursor review

File: apps/sim/app/api/function/execute/route.ts [hunk 1]
```diff
@@ -486,41 +489,37 @@ function resolveTagVariables(
...
+        try {
+          tagValue = JSON.parse(tagValue)
+        } catch {
+          // Keep as string if not valid JSON
+        }
       }
     }
 
-    const safeVarName = `__tag_${tagName.replace(/[^a-zA-Z0-9_]/g, '_')}`
+    const safeVarName = `__tag_${tagName.replace(/_/g, '_1').replace(/\./g, '_0')}`
```
[the remark this answers is not in the document]
Comment (icecrasher321) [reply apps/sim/app/api/function/execute/route.ts:522]: The current escaping scheme is actually bijective and collision-free because:
_0 in the output can only come from a . in the input
_1 in the output can only come from a _ in the input
A literal _0 in input becomes _10, not _0
A literal _1 in input becomes _11, not _1

**Comment by icecrasher321** [comment]
@cursor review
